packages/react-sdk/src/core/components/ParticipantView/ParticipantView.tsx

```diff
@@ -0,0 +1,128 @@
+import { forwardRef, ComponentType, useState, ReactElement } from 'react';
+import clsx from 'clsx';
+import {
+  SfuModels,
+  StreamVideoLocalParticipant,
+  StreamVideoParticipant,
+} from '@stream-io/video-client';
+
+import { Audio } from '../Audio';
+import { applyElementRef, Video, VideoProps } from '../Video';
+import { useTrackElementVisibility } from '../../hooks';
+import { ParticipantViewUIProps } from './DefaultParticipantViewUI';
+import { isComponentType } from '../../../utilities';
+
+export type ParticipantViewProps = {
+  /**
+   * The participant bound to this component.
+   */
+  participant: StreamVideoParticipant | StreamVideoLocalParticipant;
+
+  /**
+   * Component used to render user interface elements (details, network status...)
+   */
+  ParticipantViewUI?: ComponentType<ParticipantViewUIProps> | ReactElement;
+
+  /**
+   * In supported browsers, this sets the default audio output.
+   * The value of this prop should be a valid Audio Output `deviceId`.
+   */
+  sinkId?: string;
+
+  /**
+   * The kind of video stream to play for the given participant.
+   */
+  videoKind?: 'video' | 'screen';
+
+  /**
+   * Turns on/off the audio for the participant.
+   */
+  muteAudio?: boolean;
+
+  /**
+   * A function meant for exposing the "native" element ref to the integrators.
+   * The element can either be:
+   * - `<video />` for participants with enabled video.
+   * - `<div />` for participants with disabled video. This ref would point to
+   * the VideoPlaceholder component.
+   *
+   * @param element the element ref.
+   */
+  setVideoElementRef?: (element: HTMLElement | null) => void;
+
+  /**
+   * An additional list of class names to append to the root DOM element.
+   */
+  className?: string;
+} & Pick<VideoProps, 'VideoPlaceholder'>;
+
+export const ParticipantView = forwardRef<HTMLDivElement, ParticipantViewProps>(
+  (
+    {
+      participant,
+      sinkId,
+      videoKind = 'video',
+      muteAudio,
+      setVideoElementRef,
+      className,
+      VideoPlaceholder,
+      ParticipantViewUI,
```

Comment:
@arnautov-anton So we will keep the pattern of custom component props?

Reply:
For core, mostly logical components, yes.